src/backend/stream.rs

```diff
@@ -414,6 +414,47 @@ impl<'ctx> PulseStream<'ctx> {
             }
         }
 
+        fn get_default_sink_name(
+            mainloop: &pulse::ThreadedMainloop,
+            context: &pulse::Context,
+        ) -> String {
+            struct CallbackData<'a> {
+                mainloop: &'a pulse::ThreadedMainloop,
+                default_sink_name: String,
+            }
+
+            fn server_info_cb(
+                _context: &pulse::Context,
+                info: Option<&pulse::ServerInfo>,
+                u: *mut c_void,
+            ) {
+                let r = unsafe { ptr::NonNull::new(u as *mut CallbackData).unwrap().as_mut() };
+
+                let name = unsafe { CStr::from_ptr(info.unwrap().default_sink_name) };
+                let name_slice = name.to_bytes_with_nul();
+                r.default_sink_name = std::str::from_utf8(name_slice).unwrap().to_string();
+
+                r.mainloop.signal();
+            }
+
+            mainloop.lock();
+
+            let mut data = CallbackData {
+                mainloop,
+                default_sink_name: String::new(),
+            };
+
+            _ = context.get_server_info(server_info_cb, ptr::addr_of!(data) as *mut c_void);
+
+            mainloop.wait();
+            mainloop.unlock();
+
+            _ = data.default_sink_name.pop();
+            data.default_sink_name.push_str(".monitor");
```

Comment:
I guess the `.monitor` is the key to this magical setting. Can I have the document for this to check?

Reply:
There isn't any official page describing what ``.monitor`` does however it's use IS mentioned here and there within PulseAudio's official docs. Example being: https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/Developer/AccessControl/